modules/programs/diff-so-fancy.nix

```diff
@@ -0,0 +1,174 @@
+      '';
+    };
+
+    stripLeadingSymbols = mkOption {
+      type = types.bool;
+      default = true;
+      example = false;
+      description = ''
+        Whether the `+` or `-` at
+        line-start should be removed.
+      '';
+    };
+
+    useUnicodeRuler = mkOption {
+      type = types.bool;
+      default = true;
+      example = false;
+      description = ''
+        By default, the separator for the file header uses Unicode
+        line-drawing characters. If this is causing output errors on
+        your terminal, set this to false to use ASCII characters instead.
+      '';
+    };
+
+    rulerWidth = mkOption {
+      type = types.nullOr types.int;
+      default = null;
+      example = false;
+      description = ''
+        By default, the separator for the file header spans the full
+        width of the terminal. Use this setting to set the width of
+        the file header manually.
+      '';
+    };
+
+    enableGitIntegration = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to enable git integration for diff-so-fancy.
+
+        When enabled, diff-so-fancy will be configured as git's pager and diff filter.
+      '';
+    };
+  };
+
+  config =
+    let
+      oldOptionValue = lib.attrByPath [ "programs" "git" "diff-so-fancy" "enable" ] false config;
+      oldOptionEnabled = lib.isBool oldOptionValue && oldOptionValue;
+    in
+    lib.mkMerge [
+      (mkIf cfg.enable {
+        home.packages = [ pkgs.diff-so-fancy ];
+
+        # Auto-enable git integration if programs.git.diff-so-fancy.enable was set to true
+        programs.diff-so-fancy.enableGitIntegration = lib.mkIf oldOptionEnabled (lib.mkOverride 1490 true);
+
+        warnings =
+          lib.optional
+            (
+              cfg.enableGitIntegration && options.programs.diff-so-fancy.enableGitIntegration.highestPrio == 1490
+            )
+            "`programs.diff-so-fancy.enableGitIntegration` automatic enablement is deprecated. Please explicitly set `programs.diff-so-fancy.enableGitIntegration = true`.";
+      })
+
+      (mkIf (cfg.enable && cfg.enableGitIntegration) {
+        programs.git = {
+          enable = lib.mkDefault true;
+          iniContent =
+            let
+              dsfCommand = "${pkgs.diff-so-fancy}/bin/diff-so-fancy";
+            in
+            {
+              core.pager = "${dsfCommand} | ${pkgs.less}/bin/less ${lib.escapeShellArgs cfg.pagerOpts}";
+              interactive.diffFilter = "${dsfCommand} --patch";
+              diff-so-fancy = {
+                markEmptyLines = cfg.markEmptyLines;
+                changeHunkIndicators = cfg.changeHunkIndicators;
+                stripLeadingSymbols = cfg.stripLeadingSymbols;
+                useUnicodeRuler = cfg.useUnicodeRuler;
+                rulerWidth = mkIf (cfg.rulerWidth != null) cfg.rulerWidth;
+              };
+            };
```

Comment:
Does it have a stand-alone configuration file? Otherwise, same issue as `delta`.

Reply:
That's actually a good point. I was just blindly moving existing structure into the new modules. Didn't really analyze how each of them were set up.

Reply:
Looking at the code, I think it only supports reading Git's configuration, no stand-alone file (and no `--config`).